src/dbnode/storage/series/buffer.go

```diff
@@ -966,6 +975,22 @@ func (b *BufferBucketVersions) streams(ctx context.Context, opts streamsOptions)
 	return res
 }
 
+func (b *BufferBucketVersions) firstWriteAt(opts streamsOptions) time.Time {
+	var res time.Time
+	for _, bucket := range b.buckets {
+		if !opts.filterWriteType || bucket.writeType == opts.writeType {
```

Comment:
nit: can reduce nesting flipping this to `if bucket.writeType != opts.writeType { continue }`

Reply:
+1 w/ `if opts.filterWriteType && bucket.writeType != opts.writeType { continue }`